src/main/java/org/truffleruby/extra/ffi/Pointer.java

```diff
@@ -93,7 +93,7 @@ public void writeZeroTerminatedBytes(long offset, byte[] bytes, int start, int l
     }
 
     public void writeBytes(long offset, long size, byte value) {
-        assert address + offset != 0;
+        assert address + offset != 0 || size == 0;
```

Comment:
Is it safe to trust `UNSAFE` to no-op on 0 size? I think I'd rather handle that explicitly here.

Reply:
What else could Unsafe do?
We tell it to write zero bytes, any extra access would violate the semantics.
It's not ideal code, but the caller does `MemoryPointer.new(str.bytesize)` with `str=""` and `Unsafe.allocateMemory` returns `NULL` in such a case (allowed by the semantics of `malloc()`).

Reply:
It seems MRI and Rbx throw on all read/write operations on a null pointer, avoiding the SEGV:
```
> ptr=FFI::Pointer.new(0)
> ptr.write_array_of_int8 []
FFI::NullPointerError: invalid memory write at address=0x00000000000000
```

That sounds like a better fix, but also a much larger change which I would rather avoid in this PR.